src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1051,5 +1051,17 @@ public OptionSet Options
                 return this.Workspace.Options;
             }
         }
+
+        /// <summary>
+        /// Update current solution as a result of option changes.
+        /// 
+        /// this is a temporary workaround until editorconfig becomes real part of roslyn solution snapshot.
+        /// until then, this will explicitly fork current solution snapshot
+        /// </summary>
+        internal Solution WithOptionChanged()
+        {
+            // just create new snapshot
+            return new Solution(_state);
```

Comment:
@CyrusNajmabadi Basically, Document holds onto a snapshot of the file's options, so that way that's immutable. When the options change, we aren't (currently) creating a new snapshot so values would be stale. This is at least working around that problem before we integrate .editorconfig more cleanly into workspaces, which is currently underway.

Reply:
@jasonmalinowski What i meant was: we need to document that on the call. :)